argo/providers/vai/provider.go

```diff
@@ -225,9 +268,15 @@ func (vaip VAIProvider) CreateRun(ctx context.Context, providerConfig VAIProvide
 		return "", err
 	}
 
+	runId, err := extractPipelineNameFromSpec(pipelineJob.PipelineSpec.AsMap())
```

Comment:
I may be wrong but this strikes me as a relatively costly operation as the entire struct needs to be traveresed and converted. Could we use `GetStruct()...GetString()...` here?

Reply:
Looking

Reply:
Changed to using the struct directly to extract the value.